src/languages/regex.js

```diff
@@ -0,0 +1,101 @@
+/*
+Language: regex
+Description: Regular expressions 
+Author: Konrad Rudolph <konrad.rudolph@gmail.com>
+*/
+
+export default function (hljs) {
+  const ESCAPE_SEQUENCE = {
+    className: 'built_in',
+    begin: [
+      // Unicode script/block/category
+      /\\[pP](?:[^{]|\{\^?[^}]+\})/,
+      // Unicode character
+      /\\[ux](?:[0-9a-fA-F]{1,4}|\{[0-9a-fA-F]{1,4}\})/,
+      // Backreferences
+      /\\k(?:'-?\d+'|<-?\d+>|{-?\d+})/,
+      /\\g(?:'-?\d+'|<-?\d+>|{-?\d+}|-?\d+)/,
+      // Everything else
+      /\\./
+    ].map(r => r.source).join('|')
+  };
+
+  const CHARACTER_CLASS = {
+    className: 'string',
+    begin: /\[\^?]?/,
+    end: /]/,
+    contains: [
+      {
+        className: 'literal',
+        variants: [
+          // POSIX named character class.
+          { begin: /\[:\^?/, end: /:]/ },
```

Comment:
The question would be if something like this would get different coloring so:

```
[a-z]
[:letter]
``

The :letter is obviously quite different in context.

Reply:
Note that named POSIX classes are nested: `[a-zA-Z]` (roughly) corresponds to `[[:alpha:]]`, and `[a-zA-Z0-9]` would be `[[:alpha:][:digit:]]` (or `[[:alnum:]]`)). But really, the closer correspondence is between `[[:digit:]]` and `\d`; i.e. shorthand character classes. Doesn’t mean we have to highlight them identically, or even that we have to treat them as special anyway.